### PR TITLE
refactor: data-table

### PR DESCRIPTION
**Pull request by nekosaur**
## Description
Another rewrite of data-table. This time still using html table elements.

Implementing new functionality such as multi-column sorting, group by column, sort by column, per-column filtering, and fixed headers. Fixed headers will not be available in IE11 as I'm using `fixed` css property.

Virtualization is being done by using dummy rows prepended and appended to tbody that expand in height to simulate the full list of items.

The following issues (and more) have been implemented so far
#1547, #2659, #2859, #3102, #3176, #3180, #3574, #3486, #2960, #3829, #3220, #2890, #3364, #6138

## Markup:

https://gist.github.com/nekosaur/8ffc4996d033a71d9dbff9060eb6765c

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

File: packages/vuetify/src/components/VData/VData.ts [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+    groupDesc (groupDesc: boolean | boolean[]) {
+      this.updateOptions({ groupDesc: wrapInArray(groupDesc) })
+    },
+    'internalOptions.groupDesc' (groupDesc: boolean[], old: boolean[]) {
+      !deepEqual(groupDesc, old) && this.$emit('update:groupDesc', Array.isArray(this.groupDesc) ? groupDesc : groupDesc[0])
+    }
+  },
+
+  methods: {
+    toggle (key: string, oldBy: string[], oldDesc: boolean[], page: number, mustSort: boolean, multiSort: boolean) {
```
Comment (johnleider) [inline packages/vuetify/src/components/VData/VData.ts]: separate lines for this many args

File: packages/vuetify/src/components/VData/VDataFooter.ts [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    showNextPageIcon (): boolean {
+      return this.options.itemsPerPage < 0 ||
+        this.options.page * this.options.itemsPerPage >= this.pagination.itemsLength ||
+        this.pagination.pageStop < 0
+    }
+  },
+
+  methods: {
+    onPreviousPage () {
+      this.$emit('update:options', Object.assign({}, this.options, { page: this.options.page - 1 }))
```
Comment (johnleider) [inline packages/vuetify/src/components/VData/VDataFooter.ts]: The three `this.$emit('update:options'` could possibly be DRYed up

File: packages/vuetify/src/components/VDataTable/VRow.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      const scopedSlot = data.scopedSlots && data.scopedSlots[header.value]
+      const regularSlot = computedSlots[header.value]
+
+      if (scopedSlot) {
+        children.push(scopedSlot({ item: props.item, header, value, mobile: props.mobile }))
+      } else if (regularSlot) {
+        children.push(regularSlot)
+      } else {
+        if (props.mobile) {
+          children.push(h('div', { class: 'd-flex justify-content-between' }, [
```
Comment (johnleider) [inline packages/vuetify/src/components/VDataTable/VRow.ts]: These are helper classes and will make it hard to overwrite in userland

Comment (nekosaur) [reply packages/vuetify/src/components/VDataTable/VRow.ts]: Yup, I know. A lot of this is still work in progress and/or placeholdery

**Comment by grishnyakov** [comment]
This may be useful: multicolumn sort
https://codepen.io/grishnyakov/pen/mawOXg

**Comment by nekosaur** [comment]
multi-column sort has been implemented already

**Comment by johnleider** [comment]
Some feedback:

- Show page number
This should be in the center
image

- Custom header/footer with provide
Had multiple console errors
image

- Custom expanded example
Expand all doesn't work

- Filter by column
Had multiple console errors
image

I also was not able to figure out how to get resizable columns to work.

Other than that, everything is looking really good.

**Comment by nekosaur** [comment]
Ah, I might not have uploaded the latest playground.

**Comment by nekosaur** [comment]
@sh7dm This is far from done, no need to ask for merging.

**Comment by johnleider** [comment]
I attempted to resolve merge conflicts but there were quite a few broken tests, are you tracking this @nekosaur ?

**Comment by prune998** [comment]
Please keep on @nekosaur  !! We all need this feature !!

**Pull request merged**